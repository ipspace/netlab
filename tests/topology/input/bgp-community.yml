#
# Sample multi-vendor configuration that tests most features supported by BGP configuration module
#
# Used in manual testing (feel free to write an automated test script ;)
#
module: [ bgp,ospf ]

defaults:
  device: iosv

bgp:
  as: "65000.1"
  community:
    ibgp: [ standard, extended ]
    ebgp: standard

nodes:
  r1:
    bgp:
      community: standard
  r2:
    bgp.as: "65000.1"
    bgp:
      community: [ standard, extended ]
  r3:
    bgp.local_as: "0.65001"

links:
- r1-r3
- r2:
    bgp.local_as: 65002
  r3:
