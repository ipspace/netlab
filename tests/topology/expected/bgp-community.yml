bgp:
  advertise_loopback: true
  as: 4259840001
  community:
    ebgp:
    - standard
    ibgp:
    - standard
    - extended
  next_hop_self: true
groups:
  as4259840001:
    members:
    - r1
    - r2
    - r3
input:
- topology/input/bgp-community.yml
- package:topology-defaults.yml
links:
- interfaces:
  - ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 10.1.0.1/30
    node: r1
  - ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 10.1.0.2/30
    node: r3
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: p2p
- interfaces:
  - bgp:
      local_as: 65002
    ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 10.1.0.5/30
    node: r2
  - ifindex: 2
    ifname: GigabitEthernet0/2
    ipv4: 10.1.0.6/30
    node: r3
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 10.1.0.4/30
  type: p2p
module:
- bgp
- ospf
name: input
nodes:
  r1:
    af:
      ipv4: true
    bgp:
      advertise_loopback: true
      as: 4259840001
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        ibgp_localas:
        - standard
      ipv4: true
      neighbors:
      - activate:
          ipv4: true
        as: 4259840001
        ipv4: 10.0.0.2
        name: r2
        type: ibgp
      - activate:
          ipv4: true
        as: 4259840001
        ipv4: 10.0.0.3
        name: r3
        type: ibgp
      - activate:
          ipv4: true
        as: 65001
        ifindex: 1
        ipv4: 10.1.0.2
        name: r3
        type: ebgp
      next_hop_self: true
      router_id: 10.0.0.1
    box: cisco/iosv
    device: iosv
    id: 1
    interfaces:
    - ifindex: 1
      ifname: GigabitEthernet0/1
      ipv4: 10.1.0.1/30
      linkindex: 1
      name: r1 -> r3
      neighbors:
      - ifname: GigabitEthernet0/1
        ipv4: 10.1.0.2/30
        node: r3
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.1/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module:
    - bgp
    - ospf
    name: r1
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.1
  r2:
    af:
      ipv4: true
    bgp:
      advertise_loopback: true
      as: 4259840001
      community:
        ebgp:
        - standard
        - extended
        ibgp:
        - standard
        - extended
        ibgp_localas:
        - standard
        - extended
      ipv4: true
      neighbors:
      - activate:
          ipv4: true
        as: 4259840001
        ipv4: 10.0.0.1
        name: r1
        type: ibgp
      - activate:
          ipv4: true
        as: 4259840001
        ipv4: 10.0.0.3
        name: r3
        type: ibgp
      - activate:
          ipv4: true
        as: 65001
        ifindex: 1
        ipv4: 10.1.0.6
        local_as: 65002
        name: r3
        type: ebgp
      next_hop_self: true
      router_id: 10.0.0.2
    box: cisco/iosv
    device: iosv
    id: 2
    interfaces:
    - bgp:
        local_as: 65002
      ifindex: 1
      ifname: GigabitEthernet0/1
      ipv4: 10.1.0.5/30
      linkindex: 2
      name: r2 -> r3
      neighbors:
      - ifname: GigabitEthernet0/2
        ipv4: 10.1.0.6/30
        node: r3
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.2/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module:
    - bgp
    - ospf
    name: r2
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.2
  r3:
    af:
      ipv4: true
    bgp:
      advertise_loopback: true
      as: 4259840001
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
        ibgp_localas:
        - standard
        - extended
      ipv4: true
      local_as: 65001
      neighbors:
      - activate:
          ipv4: true
        as: 4259840001
        ipv4: 10.0.0.1
        name: r1
        type: ibgp
      - activate:
          ipv4: true
        as: 4259840001
        ipv4: 10.0.0.2
        name: r2
        type: ibgp
      - activate:
          ipv4: true
        as: 4259840001
        ifindex: 1
        ipv4: 10.1.0.1
        local_as: 65001
        name: r1
        type: ebgp
      - activate:
          ipv4: true
        as: 65002
        ifindex: 2
        ipv4: 10.1.0.5
        local_as: 65001
        name: r2
        type: ebgp
      next_hop_self: true
      router_id: 10.0.0.3
    box: cisco/iosv
    device: iosv
    id: 3
    interfaces:
    - bgp:
        local_as: 65001
      ifindex: 1
      ifname: GigabitEthernet0/1
      ipv4: 10.1.0.2/30
      linkindex: 1
      name: r3 -> r1
      neighbors:
      - ifname: GigabitEthernet0/1
        ipv4: 10.1.0.1/30
        node: r1
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - bgp:
        local_as: 65001
      ifindex: 2
      ifname: GigabitEthernet0/2
      ipv4: 10.1.0.6/30
      linkindex: 2
      name: r3 -> r2
      neighbors:
      - bgp:
          local_as: 65002
        ifname: GigabitEthernet0/1
        ipv4: 10.1.0.5/30
        node: r2
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.3/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    module:
    - bgp
    - ospf
    name: r3
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.3
ospf:
  area: 0.0.0.0
provider: libvirt
